crates/model/src/interaction.rs

```diff
@@ -1,10 +1,37 @@
+use ethcontract::Bytes;
 use primitive_types::{H160, U256};
 use serde::{Deserialize, Serialize};
 
+pub type EncodedInteraction = (
+    H160,           // target
+    U256,           // value
+    Bytes<Vec<u8>>, // callData
+);
+
+pub trait Interaction: std::fmt::Debug + Send + Sync {
```

Comment:
The model crate isn't the right place for the trait. If the type shows up in the apis then that makes sense here but not the trait. `shared` should be a better place for now.

Reply:
thanks! Sorry I missed this.

I put the trait Interaction into the shared crate now. 
But since I need the InteractionData in the models crate for the orders - since orders store interactions now, I also created a new conversion function that converts the InteractionData from a models::interaction::InteractionData to shared::interaction::InteractionData in order to use the encode functionalities.

Hope it looks better now.

Reply:
🙈 
sorry, for some reason I thought its not possible. Made no sense at all.
Thanks for highlighting